borg/archive.py

```diff
@@ -830,10 +830,10 @@ def report(msg, chunk_id, chunk_no):
                     unpacker.feed(self.key.decrypt(chunk_id, cdata))
                     try:
                         for item in unpacker:
-                            # we expect item to be a dict (and it must also work as StableDict)
-                            if isinstance(item, dict):
+                            # we expect item to be a StableDict and check if .items() works
+                            if isinstance(item, StableDict):
                                 try:
-                                    StableDict(item).items()
+                                    item.items()
```

Comment:
how about storing the valid metadata keys inside the repository, falling back to the current set as a fedault